generator/config/expressions.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+// Target namespace for the generated files, allows to use ::class notation without use statements
+
+namespace MongoDB\Builder\Expression;
+
+use MongoDB\BSON;
+use MongoDB\Model\BSONArray;
+use stdClass;
+
+/** @param class-string $resolvesTo */
+function typeFieldPath(string $resolvesTo): array
+{
+    return [
+        'class' => true,
+        'extends' => FieldPath::class,
+        'implements' => [$resolvesTo],
+        'types' => ['string'],
+    ];
+}
+
+return [
+    'null' => ['scalar' => true, 'types' => ['null']],
+    'int' => ['scalar' => true, 'types' => ['int', BSON\Int64::class]],
+    'number' => ['scalar' => true, 'types' => ['int', BSON\Int64::class]],
+    'decimal' => ['scalar' => true, 'types' => ['float', BSON\Decimal128::class]],
```

Comment:
A couple of things here:
* `double` is missing from the list; it would correspond to PHP's `float` type
* `number` can be any of the numeric types (i.e. `int`, `Int64`, `float`, `Decimal128`)
* An `int` is a valid value to be given whenever a `decimal` type is accepted.

Given that, I think this should look as follows:
```suggestion
    'int' => ['scalar' => true, 'types' => ['int', BSON\Int64::class]],
    'double' => ['scalar' => true, 'types' => ['int', BSON\Int64::class, 'float']],
    'decimal' => ['scalar' => true, 'types' => ['int', BSON\Int64::class, 'float', BSON\Decimal128::class]],
    'number' => ['scalar' => true, 'types' => ['int', BSON\Int64::class, 'float', BSON\Decimal128::class]],
```

I'll note that a 64-bit integer value may be truncated when converted to a `double`, but I believe that shouldn't be an issue in this case.

Reply:
Fixed.